From 5dd97db4e8b37ded46daca560dbf19d139f29b9d Mon Sep 17 00:00:00 2001
From: Rafael Villar Burke <pachi@ietcc.csic.es>
Date: Fri, 2 May 2025 13:25:39 +0200
Subject: [PATCH 3/3] Readd git date and commit info to version

This avoids a dependency and shows (- -) when no git repo information is found
---
 xee/build.rs    | 19 +++++++++++++++++++
 xee/src/main.rs |  4 +++-
 2 files changed, 22 insertions(+), 1 deletion(-)
 create mode 100644 xee/build.rs

diff --git a/xee/build.rs b/xee/build.rs
new file mode 100644
index 000000000..88813959b
--- /dev/null
+++ b/xee/build.rs
@@ -0,0 +1,19 @@
+use std::process::Command;
+
+fn main() {
+    let output = Command::new("git")
+        .args(["log", "-1", "--format=(%cd %h)", "--date=short"])
+        .output();
+
+    match output {
+        Ok(output) => {
+            if let Ok(git_commit) = String::from_utf8(output.stdout) {
+                println!("cargo:rustc-env=GIT_COMMIT={}", git_commit.trim());
+            }
+        }
+        Err(e) => {
+            eprintln!("Error getting git commit: {}", e);
+            println!("cargo:rustc-env=GIT_COMMIT=(- -)");
+        }
+    }
+}
diff --git a/xee/src/main.rs b/xee/src/main.rs
index 729e31930..8991d2a87 100644
--- a/xee/src/main.rs
+++ b/xee/src/main.rs
@@ -8,8 +8,10 @@ mod xpath;
 
 use clap::{Parser, Subcommand};
 
+pub(crate) const VERSION: &str = concat!(env!("CARGO_PKG_VERSION"), " ", env!("GIT_COMMIT"));
+
 #[derive(Parser)]
-#[command(author, about, version, long_about)]
+#[command(author, about, version=VERSION, long_about)]
 pub(crate) struct Cli {
     #[command(subcommand)]
     command: Commands,
-- 
2.50.0

